platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileCommon.java

```diff
@@ -345,37 +306,58 @@ public static double roundTwoDecimals(final double d) {
         return (double) Math.round(d * ROUNDING_SCALE_FACTOR) / ROUNDING_SCALE_FACTOR;
     }
 
-    public static <D> void logMergeStats(
+    public static void logCompactStats(
             final String storeName,
-            final double tookSeconds,
-            final Collection<DataFileReader<D>> filesToMerge,
+            final double tookMillis,
+            final Collection<DataFileReader<?>> filesToMerge,
             final long filesToMergeSize,
             final List<Path> mergedFiles,
-            final DataFileCollection<D> fileCollection)
+            int targetCompactionLevel,
+            final DataFileCollection<?> fileCollection)
             throws IOException {
         final long mergedFilesCount = mergedFiles.size();
         final long mergedFilesSize = getSizeOfFilesByPath(mergedFiles);
-        logger.info(
+        final double tookSeconds = tookMillis / 1000;
+        String levelsCompacted = filesToMerge.stream()
+                .map(v -> v.getMetadata().getCompactionLevel())
+                .distinct()
+                .map(v -> Integer.toString(v))
```

Comment:
Is int to String conversion really needed?

Reply:
Yes, `joining` works with `String` only.